### PR TITLE
add fill_region and draw_line to bitmap_tools

### PR DESCRIPTION
**Pull request by kmatch98**
Adds two functions to the `bitmap_tools` module:

- `fill_region` - Fills a rectangular region with a specified color value index (basically draws a rectangle)
- `draw_line` - Draws a line into a bitmap of the specified color value index

### CODE REVIEW

File: shared-bindings/bitmaptools/__init__.c [hunk 1]
```diff
@@ -243,13 +243,161 @@ STATIC mp_obj_t bitmaptools_obj_rotozoom(size_t n_args, const mp_obj_t *pos_args
 MP_DEFINE_CONST_FUN_OBJ_KW(bitmaptools_rotozoom_obj, 0, bitmaptools_obj_rotozoom);
 // requires at least 2 arguments (destination bitmap and source bitmap)
 
+//|
+//| def fill_region(
+//|        dest_bitmap: displayio.Bitmap,
+//|        x1: int, y1: int,
+//|        x2: int, y2: int,
```
Comment (tannewt) [inline shared-bindings/bitmaptools/__init__.c:250]: Are these exclusive or inclusive? I think they are generally exclusive.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question. Looks good otherwise. Thanks!

File: shared-bindings/bitmaptools/__init__.c (hunk 1 above)
Comment (jepler) [reply shared-bindings/bitmaptools/__init__.c:250]: Also, is there any requirement that `x1 <= x2` or even `x1 < x2`?

I would argue for using the same semantics as a numpy slice assignment, which in turn is an extension of how slicing works in general in Python:
```
>>> n = numpy.zeros((8,8))
>>> n[2:5, 3:7] = 1
>>> n
array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 1., 1., 1., 1., 0.],
       [0., 0., 0., 1., 1., 1., 1., 0.],
       [0., 0., 0., 1., 1., 1., 1., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]])
```
So the rows affected are 2, 3, and 4 (2 <= row < 5) and the columns affected are 3, 4, 5, and 6 (3 <= row < 7).

If `x2 <= x1` or `y2 <= y1` then it's an empty slice and no pixels are affected.

Comment (kmatch98) [reply shared-bindings/bitmaptools/__init__.c:250]: In the docstrings, I included mention that the `fill_region` function is exclusive.  
Per your suggestion, I added the "(exclusive)" comment to the `x2` and `y2` doc strings

I removed the input checks from `shared-bindings` and transferred them over into `shared-modules`.  That will help reduce the chance of mis-use.

**Review by tannewt — APPROVED** [review]
Looks good to me! Thanks!

**Pull request merged**